frame/ethereum/src/lib.rs

```diff
@@ -187,12 +188,24 @@ decl_module! {
 
 			let block = ethereum::Block {
 				header,
-				transactions,
+				transactions: transactions.clone(),
 				ommers,
 			};
 
 			BlocksAndReceipts::insert(hash, (block, receipts));
 			BlockNumbers::<T>::insert(n, hash);
+
+			for t in &transactions.clone() {
```

Comment:
If this is put above writing the block we can avoid a `clone`. Not a big issue though.